clang/lib/CIR/CodeGen/CIRGenDecl.cpp

```diff
@@ -471,7 +471,7 @@ CIRGenModule::getOrCreateStaticVarDecl(const VarDecl &D,
   mlir::Attribute Init = nullptr;
   if (Ty.getAddressSpace() == LangAS::opencl_local ||
       D.hasAttr<CUDASharedAttr>() || D.hasAttr<LoaderUninitializedAttr>())
-    llvm_unreachable("OpenCL & CUDA are NYI");
+    Init = builder.getUndefAttr(getTypes().ConvertType(Ty));
```

Comment:
@bcardosolopes I see. On LLVM side, ppl used something like this to setup the `__shared__` variables. For the same purpose, can we simply remove this `llvm_unreachable` and also remove the line that setting the initializer?

Reply:
That sounds good. Please also add a C/C++ source code test that goes all the way down to LLVM, there are other examples you can look (e.g. `clang/test/CIR/CodeGen/atomic.cpp`)